crypter.go

```diff
@@ -489,7 +496,9 @@ func (obj JSONWebEncryption) Decrypt(decryptionKey interface{}) ([]byte, error)
 		plaintext, err = cipher.decrypt(cek, authData, parts)
```

Comment:
This err winds up never getting checked, in the "compression" case!

I think the logic here needs to become more idiomatic. In this function, since there's only a single possible recipient, we should change to `if err != nil { return nil, err }`. And do the same to the result of the decrypt call. That way we are relying on `err != nil` as our error condition rather than `plaintext == nil`. Which is important now that we realize plaintext _can_ be nil after a valid decryption.

Below in DecryptMulti it's a little messier. We should add a `var found bool`, and for each operation in the loop across recipients, we should say `if err != nil { continue }`; and in the alternate case, set `found = true` and `break`. Then instead of checking `plaintext == nil` we can check `found`.